### PR TITLE
Reduce use of downcast<>() in rendering/style code

### PR DESCRIPTION
**Pull request by cdumez**
#### 97b4d91001b8f19922b9c80bdf8000305d1996aa
<pre>
Reduce use of downcast&lt;&gt;() in rendering/style code
<a href="https://bugs.webkit.org/show_bug.cgi?id=266830">https://bugs.webkit.org/show_bug.cgi?id=266830</a>

Reviewed by Tim Nguyen.

* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::BasicShapeCircle::operator== const):
(WebCore::BasicShapeCircle::canBlend const):
(WebCore::BasicShapeEllipse::operator== const):
(WebCore::BasicShapeEllipse::canBlend const):
(WebCore::BasicShapeRect::operator== const):
(WebCore::BasicShapeXywh::operator== const):
(WebCore::BasicShapePolygon::operator== const):
(WebCore::BasicShapePolygon::canBlend const):
(WebCore::BasicShapePath::operator== const):
(WebCore::BasicShapePath::canBlend const):
(WebCore::BasicShapeInset::operator== const):
* Source/WebCore/rendering/style/ContentData.h:
(WebCore::operator==):
* Source/WebCore/rendering/style/GridPositionsResolver.cpp:
(WebCore::adjustGridPositionsFromStyle):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::setContent):
* Source/WebCore/rendering/style/StyleCachedImage.cpp:
(WebCore::StyleCachedImage::operator== const):
* Source/WebCore/rendering/style/StyleCanvasImage.cpp:
(WebCore::StyleCanvasImage::operator== const):
* Source/WebCore/rendering/style/StyleCrossfadeImage.cpp:
(WebCore::StyleCrossfadeImage::operator== const):
* Source/WebCore/rendering/style/StyleCursorImage.cpp:
(WebCore::StyleCursorImage::operator== const):
(WebCore::StyleCursorImage::selectBestFitImage):
(WebCore::StyleCursorImage::updateCursorElement):
* Source/WebCore/rendering/style/StyleCursorImage.h:
* Source/WebCore/rendering/style/StyleFilterImage.cpp:
(WebCore::StyleFilterImage::operator== const):
(WebCore::StyleFilterImage::load):
* Source/WebCore/rendering/style/StyleGradientImage.cpp:
(WebCore::StyleGradientImage::operator== const):
* Source/WebCore/rendering/style/StyleImageSet.cpp:
(WebCore::StyleImageSet::operator== const):
* Source/WebCore/rendering/style/StyleMultiImage.cpp:
(WebCore::StyleMultiImage::load):
* Source/WebCore/rendering/style/StyleNamedImage.cpp:
(WebCore::StyleNamedImage::operator== const):
* Source/WebCore/rendering/style/StylePaintImage.cpp:
(WebCore::StylePaintImage::operator== const):

Canonical link: <a href="https://commits.webkit.org/272481@main">https://commits.webkit.org/272481@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/c0177fc650cb98c9fc1cc7241df8d6bc72810cc1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31791 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10482 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33529 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34292 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28789 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32585 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12842 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7724 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28385 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32153 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8843 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28388 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7634 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7804 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28301 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35639 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28907 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28753 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/33916 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7893 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5891 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/31775 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9547 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7449 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8564 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8415 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/rendering/style/RenderStyle.cpp [hunk 1]
```diff
@@ -2239,12 +2239,10 @@ void RenderStyle::setContent(const String& string, bool add)
     auto& data = m_nonInheritedData.access().miscData.access();
     if (add && data.content) {
         auto& last = lastContent(*data.content);
-        if (!is<TextContentData>(last))
+        if (auto* textContent = dynamicDowncast<TextContentData>(last))
+            textContent->setText(textContent->text() + string);
+        else
```
Comment (nt1m) [inline Source/WebCore/rendering/style/RenderStyle.cpp]: I removed this code, please make sure you undo this change when you rebase.

**Review by nt1m — APPROVED** [review]

File: Source/WebCore/rendering/style/RenderStyle.cpp (hunk 1 above)
Comment (cdumez) [reply Source/WebCore/rendering/style/RenderStyle.cpp]: Thanks for the heads-up. Rebased.
